src/branchIfFalsy.tsx

```diff
@@ -0,0 +1,31 @@
+import {ReactElement} from 'react'
+import {branch, renderNothing, returns} from 'ad-hok'
+import {some} from 'lodash/fp'
+
+import {ensureArray} from './utils'
+
+type NonFalse<T> = T extends boolean ? T & true : T
+
+type BranchIfFalsyType = <TProps extends {}, TPropNames extends keyof TProps>(
```

Comment:
Super-minor nitpick, but I think type names like `TPropNames` should be singular. (eg. `string` is the type of all strings, not `strings`, so `TPropName` should be the type of all prop names)

Reply:
Sure, updated to `TPropName`